Automattic/wp-calypso
# Add WP forum topic block

#### Proposed Changes

* Add new GB block `forum-topic` to allow embedding WP.com forum topics. Implementation is shared with `support-page` block. Design in 111-gh/Automattic/lighthouse-forums.

Things that will be implemented in a separate PR:
- Display the topic status
- Display the topic author
To display these things, we'll need to update the data that WP API returns for topics.

Editor screenshot:
![image](https://user-images.githubusercontent.com/133608/203081830-e63df588-6f35-40c4-8293-db011e2f9b47.png)

Rendered content screenshot:
![image](https://user-images.githubusercontent.com/133608/203081959-4ab63327-8886-4d8a-86de-d83e57c689c2.png)

 
#### Testing Instructions

* Run `yarn dev --sync` in `apps/happy-blocks` to sync code to your sandbox
* On your sandbox, edit `wp-content/mu-plugins/wpcom-bbpress.php` and add these lines
```
		if ( file_exists( WP_CONTENT_DIR . '/a8c-plugins/happy-blocks/index.php' ) ) {
			require_once WP_CONTENT_DIR . '/a8c-plugins/happy-blocks/index.php';
		}
```
to the function `wpcom_bbpress_load_plugins` to load `happy-blocks` plugin.
* On your sandbox, edit `wp-content/themes/a8c/supportforums/editor/gutenberg-wpcom.php` and add to the function `supportforums_gutenberg_setup_frontend` before `return $settings;` line:
```
$settings['iso']['blocks']['allowBlocks'][] = 'happy-blocks/support-page';
$settings['iso']['blocks']['allowBlocks'][] = 'happy-blocks/forum-topic';
```
* Create a new topic, in the editor press `/` and search for "WordPress.com Forums" block.
* Insert *public* forum topic URL, for example, https://wordpress.com/forums/topic/request-a-quote-on-menu/, and verify its content has been embedded in the editor.
* Save the topic, and verify embedded block was rendered in the topic view.
* Verify embed is inserted by pasting the forum topic URL directly into the editor.
* Verify topic creation date is updated without re-editing.
* Verify support page embed works.


Note, that only public forums topics could be embedded.

#### Pre-merge Checklist





Related to #
